manifest/morph/morph.go

```diff
@@ -23,6 +22,15 @@ func ValueToType(v tftypes.Value, t tftypes.Type, p *tftypes.AttributePath) (tft
 		return tftypes.Value{}, diags
 	}
 	if v.IsNull() {
+		if err := tftypes.ValidateValue(t, nil); err != nil {
```

Comment:
This whole block of code is repeated verbatim in quite a few places. Can you think of a way where we could capture this logic once but re-use it in each of these places?